lib/recording/RecordingEntrySession.js

```diff
@@ -358,27 +375,58 @@ class RecordingEntrySession {
                 }
                 return this.updateJson(Mp4FilesUpdated)
             })
-            .then(()=>{
+            .then(()=> {
                 return this.updateEntryDuration()
             })
-            .catch((err) =>{
+            .then(()=> {
+                let currentTime = new Date();
+                return this.checkIfSplitSession(currentTime);
+            })
+            .catch((err) => {
                 this.logger.error("Error occurred while trying to add new chunks: %s", ErrorUtils.error2string(err));
                 // Only method that calls this function is linkAndUpdateJson (above), and serializedActionsPromise cannot be rejected.
                 return Q.resolve();
             })
 
     }
 
-    changeRecordedEntry(recordedEntryId){
+    changeRecordedEntry(recordedEntryId) {
+        // Todo: decide to use only one member containg recordedEntryId be it the this.entryObject or the this.recordedEntryId
         this.recordedEntryId = recordedEntryId;
+        this.entryObject.recordedEntryId = recordedEntryId;
```

Comment:
sounds like duplication... we should keep and reference only one recordedEntryId value (in entryObject)